src/Molecules.cpp

```diff
@@ -45,7 +46,12 @@ void Molecules::Init(Setup & setup, Forcefield & forcefield,
   kindsCount = setup.mol.kindMap.size();
   countByKind = new uint[kindsCount];
   kinds = new MoleculeKind[kindsCount];
-
+  assert(kindsCount == setup.mol.molVars.molKindIndex);
```

Comment:
It would exit here and would not print the error message that you have in next line

Reply:
Removed assertion.